app/views/results/common/_annotations.js.erb

```diff
@@ -380,8 +380,9 @@
     positions = null;
     annotation_manager = null;
 
+    // TODO - Need to see if submission_id is required here, or if it breaks something unexpectedly?
     jQuery.ajax({
-      url:  '<%= codeviewer_assignment_submission_result_path(uid: @uid) %>',
+      url:  '<%= codeviewer_assignment_submission_result_path(uid: @uid, submission_id: @submission.id) %>',
```

Comment:
This is a place where the path helper should be different depending on whether it's student or not.

I.e., there needs to be a `codeviewer_assignment_result_path` helper, corresponding to a codeviewer route under `assignments/../results/`